### PR TITLE
CFY-3682 - Add implementation for authrization based on roles

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

File: new/simple-manager-blueprint.yaml [hunk 1]
```diff
@@ -346,6 +346,9 @@ node_types:
   cloudify.nodes.ManagerHost:
     derived_from: cloudify.nodes.Compute
     properties:
+      manager_env_base_dir:
+        type: string
+        default: /opt/manager
```
Comment (noakup) [inline new/simple-manager-blueprint.yaml]: add in Openstack and AWS as well.

File: new/resources/rest/roles_config.yaml [hunk 2]
```diff
@@ -1,141 +1,79 @@
 administrator:
```
Comment (noakup) [inline new/resources/rest/roles_config.yaml]: add explanation in the file as a comment

File: new/resources/rest/roles_config.yaml [hunk 3]
```diff
@@ -1,141 +1,83 @@
+# This file maps roles to allowed and denied permissions.
+# Access will be granted only to allowed permissions, that do not appear in the deny dictionary.
+# Each permission is a dict of <endpoint>:<http methods as list>
+
 administrator:
-  - 'v2/blueprints/*':
```
Comment (dankilman) [inline new/resources/rest/roles_config.yaml]: keep `admin..` and keep rest but commented out

File: new/types/manager-types.yaml [hunk 4]
```diff
@@ -85,9 +86,6 @@ node_types:
             implementation: flask_securest.userstores.simple:SimpleUserstore
             properties:
               userstore:
-                user:
-                  username: { get_input: admin_username }
-                  password: { get_input: admin_password }
                 users:
                   - username: { get_input: admin_username }
                     password: { get_input: admin_password }
```
Comment (dankilman) [inline new/types/manager-types.yaml]: remove other users

File: new/types/manager-types.yaml [hunk 5]
```diff
@@ -71,6 +71,7 @@ node_types:
           authorization_provider:
             implementation: flask_securest.authorization_providers.role_based_authorization_provider:RoleBasedAuthorizationProvider
             properties:
+              roles_config_file_path: 'opt/manager/roles_config.yaml'
```
Comment (dankilman) [inline new/types/manager-types.yaml]: add `/`

**Pull request merged**